File: app/src/main/java/com/canli/oya/traininventory/adapters/CustomSpinAdapter.java
package com.canli.oya.traininventory.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.canli.oya.traininventory.R;
import com.canli.oya.traininventory.data.entities.BrandEntry;
import com.canli.oya.traininventory.data.entities.CategoryEntry;
import com.canli.oya.traininventory.utils.GlideApp;

import java.util.List;

public class CustomSpinAdapter extends BaseAdapter {

    private Context mContext;
    private List<BrandEntry> mBrandList;

    public CustomSpinAdapter(Context context, List<BrandEntry> list) {
        mContext = context;
        mBrandList = list;
    }

    @Override
    public int getCount() {
        return mBrandList.size();
    }

    @Override
    public BrandEntry getItem(int position) {
        return mBrandList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(mContext).
                    inflate(R.layout.spinner_item, parent, false);
        }

        BrandEntry currentBrand = getItem(position);

        TextView brandName_tv = convertView.findViewById(R.id.spin_item_brand_name);
        ImageView logo_iv = convertView.findViewById(R.id.spin_item_logo);

        brandName_tv.setText(currentBrand.getBrandName());
        GlideApp.with(mContext)
                .load(currentBrand.getBrandLogoUri())
                .centerCrop()
                .placeholder(R.drawable.placeholder)
                .into(logo_iv);

        return convertView;
    }
}

File: app/src/main/java/com/canli/oya/traininventoryroom/adapters/CustomSpinAdapter.kt
package com.canli.oya.traininventoryroom.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.canli.oya.traininventoryroom.R
import com.canli.oya.traininventoryroom.data.BrandEntry
import com.canli.oya.traininventoryroom.utils.GlideApp

class CustomSpinAdapter(private val mContext: Context, var mBrandList: List<BrandEntry>?) : BaseAdapter() {

    override fun getCount(): Int {
        return mBrandList?.size ?: 0
    }

    override fun getItem(position: Int): BrandEntry? {
        return mBrandList?.get(position)
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, view: View?, parent: ViewGroup): View {
        return createViewFromResource(view, parent, position, R.layout.item_spinner)
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        return createViewFromResource(convertView, parent, position, R.layout.item_spinner_dropdown)
    }

    private fun createViewFromResource(view: View?, parent: ViewGroup, position: Int, resource : Int): View {
        var convertView = view
        if (convertView == null) {
            convertView = LayoutInflater.from(mContext).inflate(resource, parent, false)
        }

        val currentBrand = getItem(position)

        val brandName = convertView!!.findViewById<TextView>(R.id.spin_item_brand_name)
        val logo = convertView.findViewById<ImageView>(R.id.spin_item_logo)

        brandName.text = currentBrand?.brandName
        val imageUri = currentBrand?.brandLogoUri
        if (imageUri == null) {
            logo.visibility = View.GONE
        } else {
            logo.visibility = View.VISIBLE
            GlideApp.with(mContext)
                    .load(imageUri)
                    .centerCrop()
                    .into(logo)
        }

        return convertView
    }
}
